mlir/lib/ExecutionEngine/ROCm/BackendUtils.cpp

```diff
@@ -196,7 +196,7 @@ OwnedBlob BackendUtils::compileISAToHsaco(const std::string isa, Location loc,
 std::unique_ptr<llvm::Module> BackendUtils::compileModuleToROCDLIR(
     Operation *m, llvm::LLVMContext &llvmContext, llvm::StringRef name) {
   StringRef amdgcnDataLayout =
-      "e-p:64:64-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-i64:64-"
+      "e-p:32:32-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-i64:64-"
```

Comment:
so global pointer addressing is back to 32 bit?

Reply:
yes. I find doing so could help get rid of quite a few `ashr` instruction in the final ISA.
but maybe let me remove this `i32/i64` feature from this PR, and make it a standalone one.